pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -357,36 +372,36 @@ func (config *V2RegistriesConf) postProcessRegistries() error {
 			return err
 		}
 
-		if reg.Prefix == "" {
-			if reg.Location == "" {
-				return &InvalidRegistries{s: "invalid condition: both location and prefix are unset"}
-			}
-			reg.Prefix = reg.Location
-		} else {
-			reg.Prefix, err = parseLocation(reg.Prefix)
+		if reg.Prefix != "" {
+			reg.Prefix, err = parsePrefix(reg.Prefix)
 			if err != nil {
 				return err
 			}
-			// FIXME: allow config authors to always use Prefix.
-			// https://github.com/containers/image/pull/1191#discussion_r610622495
-			if reg.Prefix[:2] != "*." && reg.Location == "" {
-				return &InvalidRegistries{s: "invalid condition: location is unset and prefix is not in the format: *.example.com"}
+			if reg.Location != "" {
```

Comment:
This looks suspicious.  Why are we allowing a config with an empty location?

If that's desired, we need tests for that case and probably some sanity checks and some massaging of the docs.  For instance, unless it's blocked, it would require at least one mirror to make sense which in turn would bring up the question: why lift the requirement to specify a location,

Reply:
See all the “prefix is a wildcard” and “prefix is empty” conditions in https://github.com/openshift/runtime-utils/pull/11/files `EditRegistriesConfig`.

Right now, if the user wants to set properties (insecure/blocked/mirrors) on a registry _without redirecting it_, there are two different syntaxes: (prefix=*.domain, location="") or (prefix="", location="non-wildcard.domain/…"), with little reason to for the two to differ. It would conceptually be much simpler if we always required `prefix` to exist as a lookup key, and made `location` entirely optional for redirection only; we can’t do that any more due to compatibility, but at least _allowing_ software that writes entries from scratch to always use `prefix` could simplify it.